src/main/java/ru/hogwarts/school/controllers/StudentController.java

```diff
@@ -89,5 +91,41 @@ public ResponseEntity<Student> deleteStudent(@PathVariable Long id) {
         return ResponseEntity.ok(student);
     }
 
+    @GetMapping("/print-parallel")
+    public void printStudentsInParallel() {
+        List<Student> studentList = new ArrayList<>(getAllStudents());
+        System.out.println(studentList.get(0).getName());
+        System.out.println(studentList.get(1).getName());
+        new Thread(() -> {
+            System.out.println(studentList.get(2).getName());
+            System.out.println(studentList.get(3).getName());
+        }).start();
+        new Thread(() -> {
+            System.out.println(studentList.get(4).getName());
+            System.out.println(studentList.get(5).getName());
+        }).start();
+    }
+
+    private final Object studentSynchronizer = new Object();
+
+    @GetMapping("/print-synchronized")
+    public void printStudentsSynchronized() {
+        List<Student> studentList = new ArrayList<>(getAllStudents());
+        synchronizedMethod(studentList.get(0));
+        synchronizedMethod(studentList.get(1));
+        new Thread(() -> {
+            synchronizedMethod(studentList.get(2));
+            synchronizedMethod(studentList.get(3));
+        }).start();
+        new Thread(() -> {
+            synchronizedMethod(studentList.get(4));
+            synchronizedMethod(studentList.get(5));
+        }).start();
+    }
 
+    public void synchronizedMethod(Student student) {
```

Comment:
Можно было бы прям метод синхронизированный сделать, работало бы точно так же, так как у тебя весь блок метода синхронизирован, но без дополнительного монитора
```suggestion
    public synchronized void synchronizedMethod(Student student) {
```